src/main/java/teammates/logic/core/LoggingLogic.java

```diff
@@ -0,0 +1,84 @@
+package teammates.logic.core;
+
+import java.time.Instant;
+import java.util.List;
+
+import com.google.cloud.MonitoredResource;
+import com.google.cloud.logging.LogEntry;
+import com.google.cloud.logging.Payload.StringPayload;
+
+import teammates.common.exception.LoggingServiceException;
+import teammates.common.util.Config;
+import teammates.common.util.Logger;
+import teammates.logic.core.LoggingService.LogSearcher;
+
+/**
+ * Handles the logic related to logging.
+ */
+public final class LoggingLogic {
```

Comment:
I think `LogsLogic` might be more appropriate, as it's the logic for managing logs (the noun). Just an opinion though so willing to hear what others might think